plugins/bkpr/db.c

```diff
@@ -0,0 +1,168 @@
+#include "config.h"
+#include <ccan/array_size/array_size.h>
+#include <db/bindings.h>
+#include <db/common.h>
+#include <db/exec.h>
+#include <db/utils.h>
+#include <plugins/bkpr/db.h>
+#include <plugins/libplugin.h>
+#include <stdio.h>
+
+struct migration {
+	const char *sql;
+	void (*func)(struct plugin *p, struct db *db);
+};
+
+/* Do not reorder or remove elements from this array.
+ * It is used to migrate existing databases from a prevoius state, based on
+ * string indicies */
+static struct migration db_migrations[] = {
+	{SQL("CREATE TABLE version (version INTEGER);"), NULL},
```

Comment:
Probably a tall ask, but why a separate database?  Could we use `datastore` instead?

Reply:
iiuc `datastore` is designed for key-value storage; this application requires more complex relational data storage.

Reply:
All relational data storage is based on key-value ultimately (tables have a row ID per row that is the key to the tuple, and indices map tuple values to row ID keys), though I do understand wanting not to have to reimplement SQL when a perfectly good library already exists.

In particular, this seems to be recordkeeping that is "safe" to lose; you lose your records, but none of your funds are at risk if the `bookkeeper`-specific database is lost due to inevitable hardware failure.  But I think you should also point this out in the docs, then, that since it is not stored in `lightningd.sqlite3` or in the PostgreSQL  database, then any automated backup/replication setup the user uses does ***not*** apply to the `bookkeeper` data.